lte/gateway/docker/docker-compose.yaml

```diff
@@ -94,45 +110,78 @@ services:
   eventd:
     <<: *pyservice
     container_name: eventd
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "50075"]
+      timeout: "4s"
+      retries: 3
     command: /usr/bin/env python3 -m magma.eventd.main
 
   health:
     <<: *pyservice
     container_name: health
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "50080"]
+      timeout: "4s"
+      retries: 3
     # Needed in order to enable/disable ICMP
     privileged: true
     command: /usr/bin/env python3 -m magma.health.main
 
   monitord:
     <<: *pyservice
     container_name: monitord
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "50076"]
+      timeout: "4s"
+      retries: 3
     command: /usr/bin/env python3 -m magma.monitord.main
 
   redirectd:
     <<: *pyservice
     container_name: redirectd
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "50071"]
+      timeout: "4s"
+      retries: 3
     command: /usr/bin/env python3 -m magma.redirectd.main
 
   smsd:
     <<: *pyservice
     container_name: smsd
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "50078"]
+      timeout: "4s"
+      retries: 3
     command: /usr/bin/env python3 -m magma.smsd.main
 
   control_proxy:
     <<: *pyservice
     container_name: control_proxy
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "8443"]
+      timeout: "4s"
+      retries: 3
     command: >
       sh -c "/usr/local/bin/generate_nghttpx_config.py && sed -i 's/errorlog-syslog=yes/errorlog-syslog=no/' /var/opt/magma/tmp/nghttpx.conf &&\
              /usr/bin/env nghttpx --conf /var/opt/magma/tmp/nghttpx.conf /var/opt/magma/certs/controller.key /var/opt/magma/certs/controller.crt"
 
   ctraced:
     <<: *pyservice
     container_name: ctraced
+    healthcheck:
+      test: ["CMD", "nc", "-zv", "localhost", "50079"]
+      timeout: "4s"
+      retries: 3
     command: /usr/bin/env python3 -m magma.ctraced.main
 
   sctpd:
     <<: *ltecservice
     container_name: sctpd
+    # Figure out sctp test
+    # healthcheck:
+    #   test: ["CMD", "nc", "-zv", "localhost", "50067"]
+    #   timeout: "4s"
+    #   retries: 3
```

Comment:
why is the sctpd health check is not enabled ?

Reply:
sctpd afaik doesn't listen on any tcp/udp port we can check. I couldn't figure out a reliable way to test for port 36412 or some other health check for sctpd.

```
root@ip-10-7-4-43:/var/opt/magma/tmp# nc -vz localhost 36412
nc: connect to localhost port 36412 (tcp) failed: Connection refused
root@ip-10-7-4-43:/var/opt/magma/tmp# nc -vz 10.7.2.67 36412
nc: connect to 10.7.2.67 port 36412 (tcp) failed: Connection refused
root@ip-10-7-4-43:/var/opt/magma/tmp# ps -ef
UID          PID    PPID  C STIME TTY          TIME CMD
root           1       0  0 Oct14 ?        00:06:47 /usr/local/bin/sctpd
root         199       0  0 19:08 pts/0    00:00:00 bash
root         216     199  0 19:09 pts/0    00:00:00 ps -ef
root@ip-10-7-4-43:/var/opt/magma/tmp# lsof -p 1 | grep LISTEN
root@ip-10-7-4-43:/var/opt/magma/tmp#
```

I thought of checking for the sctp active endpoint, but that's just OS level check, not application level.
```
root@ip-10-7-4-43:/var/opt/magma/tmp# cat /proc/net/sctp/eps
 ENDPT     SOCK   STY SST HBKT LPORT   UID INODE LADDRS
ffff98e4a0dc7c00 ffff98e488408600 2   10  9    36412     0 1779729 10.7.2.67
```

Do you have any tricks for checking sctpd health?

Reply:
@rmeleromira Yes, we are using unix socket for sctpd.
You can use the following,

root@agw:/var/opt/magma/docker# ncat -vz -u --sctp -4 -U /tmp/sctpd_upstream.sock 
Ncat: Version 7.80 ( https://nmap.org/ncat )
Ncat: Connected to /tmp/sctpd_upstream.sock.
Ncat: 0 bytes sent, 0 bytes received in 0.01 seconds.

root@agw:/var/opt/magma/docker# ncat -vz -u --sctp -4 -U /tmp/sctpd_downstream.sock 
Ncat: Version 7.80 ( https://nmap.org/ncat )
Ncat: Connected to /tmp/sctpd_downstream.sock.
Ncat: 0 bytes sent, 0 bytes received in 0.01 seconds.

Reply:
A future task for the k8s effort in Q1 would be to eliminate some of these dependencies like Unix sockets. Given we will move away from unix sockets soon, this should be something that we can mark as a known issue.

Reply:
> 

Reported issue https://github.com/magma/magma/issues/10573 , I'm not familiar with the effort to remove unix sockets so issue is kind of anemic.